core/sparqlbuilder/src/test/java/org/eclipse/rdf4j/sparqlbuilder/examples/sparql11spec/Section6Test.java

```diff
@@ -61,4 +61,19 @@ public void example_6_3() {
 						GraphPatterns.and(x.has(foaf.iri("homepage"), hpage)).optional());
 		p();
 	}
+
+	@Test
+	public void example_6_4() {
+		Prefix dc = SparqlBuilder.prefix("dc", iri(DC_NS)), ns = SparqlBuilder.prefix("ns", iri(EXAMPLE_ORG_NS));
+		Variable title = SparqlBuilder.var("title"), price = SparqlBuilder.var("price"), x = SparqlBuilder.var("x");
+
+		GraphPatternNotTriples pricePattern = GraphPatterns.and(x.has(ns.iri("price"), price))
+				.filter(Expressions.or(Expressions.lt(price, 20),
+						Expressions.and(Expressions.lt(price, 50), Expressions.gt(price, 30))))
+				.optional();
+
+		query.prefix(dc, ns).select(title, price).where(x.has(dc.iri("title"), title), pricePattern);
+		p();
+	}
+
```

Comment:
I don't see how this tests actually checks if the changes you've made are correct. The test still passes even if I comment out your change.

Reply:
Btw. I'm sorry the test code in general here is so poor. `p()` isn't particularly useful here (for those reading this comment it actually means System.out.println(query) ), and all the tests are stateful, even though there is no need to keep state between tests.

Reply:
Second btw. I've take another look at the tests within the `examples` package and realises that none of these test anything, ie. there are no asserts anywhere. When I remove these "tests" I end up with a 13% test coverage. This is very unfortunate, but in now way your fault @abhishekag03

Reply:
I propose this assert:

```
Assert.assertThat(query.getQueryString(), CoreMatchers.containsString("( ?price < 50 && ?price > 30 )"));
```